### PR TITLE
fix: add external non secure mqtt broker for taf testing

### LINKED ISSUE #287: External NonSecure MQTT broker for taf testing
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]

 This feature request is for taf testing requires a non secure mqtt broker when conducting some checks.

### Description [**REQUIRED**]
 A clear and concise description of the problem or missing capability...
As of now the default mqtt broker is in secure mode and taf testing scripts are not able to connect.

### Describe the solution you'd like
 If you have a solution in mind, please describe it.
Add external mqtt broker for taf only

### Describe alternatives you've considered
 Have you considered any alternative solutions or workarounds?

### PR DESCRIPTION
**Pull request by joshua-silverio**
Closes #287
Signed-off-by: intel <joshua.silverio@intel.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

- Make build-taf
- replace CONF_DIR_PLACE_HOLDER -> /custom-config
- replace MQTT_BROKER_ADDRESS_PLACE_HOLDER -> tcp://edgex-mqtt-taf-broker:1884
- docker-compose -f ../taf/composefile.yml up -d
- try to connect to secure broker edgex-mqtt-broker:1883 (should be not authorized)
- try to connect to edgex-mqtt-taf-broker:1884 (should be allowed as non secure mode)

### CODE REVIEW

File: taf/docker-compose-taf-mqtt-bus-arm64.yml [hunk 1]
```diff
@@ -78,10 +78,10 @@ services:
       TRIGGER_EDGEXMESSAGEBUS_SUBSCRIBEHOST_HOST: edgex-redis
       TRIGGER_EXTERNALMQTT_AUTHMODE: usernamepassword
       TRIGGER_EXTERNALMQTT_SECRETPATH: message-bus
-      TRIGGER_EXTERNALMQTT_URL: tcp://MQTT_BROKER_ADDRESS_PLACE_HOLDER:1883
+      TRIGGER_EXTERNALMQTT_URL: MQTT_BROKER_ADDRESS_PLACE_HOLDER
       WRITABLE_LOGLEVEL: INFO
       WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_AUTHMODE: usernamepassword
-      WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_BROKERADDRESS: tcp://MQTT_BROKER_ADDRESS_PLACE_HOLDER:1883
+      WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_BROKERADDRESS: MQTT_BROKER_ADDRESS_PLACE_HOLDER
       WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_SECRETPATH: message-bus
```
Comment (lenny-intel) [inline taf/docker-compose-taf-mqtt-bus-arm64.yml:85]: This line should be removed

File: taf/docker-compose-taf-mqtt-bus-arm64.yml [hunk 2]
```diff
@@ -291,7 +291,7 @@ services:
       WRITABLE_INSECURESECRETS_MQTT_SECRETS_USERNAME: USERNAME_PLACEH_OLDER
       WRITABLE_LOGLEVEL: INFO
       WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_AUTHMODE: usernamepassword
```
Comment (lenny-intel) [inline taf/docker-compose-taf-mqtt-bus-arm64.yml:293]: Same

File: taf/docker-compose-taf-mqtt-bus-arm64.yml [hunk 3]
```diff
@@ -291,7 +291,7 @@ services:
...
-      WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_BROKERADDRESS: tcp://MQTT_BROKER_ADDRESS_PLACE_HOLDER:1883
+      WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_BROKERADDRESS: MQTT_BROKER_ADDRESS_PLACE_HOLDER
       WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_SECRETPATH: message-bus
```
Comment (lenny-intel) [inline taf/docker-compose-taf-mqtt-bus-arm64.yml:295]: Same

File: taf/docker-compose-taf-mqtt-bus-arm64.yml [hunk 4]
```diff
@@ -78,10 +78,10 @@ services:
       TRIGGER_EDGEXMESSAGEBUS_SUBSCRIBEHOST_HOST: edgex-redis
       TRIGGER_EXTERNALMQTT_AUTHMODE: usernamepassword
       TRIGGER_EXTERNALMQTT_SECRETPATH: message-bus
```
Comment (lenny-intel) [inline taf/docker-compose-taf-mqtt-bus-arm64.yml:80]: Should be removed

File: taf/docker-compose-taf-mqtt-bus-arm64.yml [hunk 5]
```diff
@@ -78,10 +78,10 @@ services:
...
-      TRIGGER_EXTERNALMQTT_URL: tcp://MQTT_BROKER_ADDRESS_PLACE_HOLDER:1883
+      TRIGGER_EXTERNALMQTT_URL: MQTT_BROKER_ADDRESS_PLACE_HOLDER
       WRITABLE_LOGLEVEL: INFO
       WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_AUTHMODE: usernamepassword
```
Comment (lenny-intel) [inline taf/docker-compose-taf-mqtt-bus-arm64.yml:83]: Should be removed

File: taf/docker-compose-taf-mqtt-bus-arm64.yml [hunk 6]
```diff
@@ -1380,7 +1394,7 @@ services:
       WRITABLE_INSECURESECRETS_MQTT_SECRETS_USERNAME: USERNAME_PLACEH_OLDER
       WRITABLE_LOGLEVEL: DEBUG
       WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_AUTHMODE: usernamepassword
```
Comment (lenny-intel) [inline taf/docker-compose-taf-mqtt-bus-arm64.yml:1396]: Same

File: taf/docker-compose-taf-mqtt-bus-arm64.yml [hunk 7]
```diff
@@ -1380,7 +1394,7 @@ services:
...
-      WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_BROKERADDRESS: tcp://MQTT_BROKER_ADDRESS_PLACE_HOLDER:1883
+      WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_BROKERADDRESS: MQTT_BROKER_ADDRESS_PLACE_HOLDER
       WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_CLIENTID: scalability-test-mqtt-export
       WRITABLE_PIPELINE_FUNCTIONS_MQTTEXPORT_PARAMETERS_SECRETPATH: message-bus
```
Comment (lenny-intel) [inline taf/docker-compose-taf-mqtt-bus-arm64.yml:1399]: Same

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: taf/docker-compose-taf-mqtt-bus-arm64.yml (hunk 1 above)
Comment (joshua-silverio) [reply taf/docker-compose-taf-mqtt-bus-arm64.yml:85]: sorry, which line are you referring to in these comments

Comment (lenny-intel) [reply taf/docker-compose-taf-mqtt-bus-arm64.yml:85]: Authmode and SecretPath

**Review by lenny-intel — APPROVED** [review]
LGTM, nice job!

**Review by bnevis-i — COMMENTED** [review]
Waive review.  Looks non-malicious. :-)

**Review by cherrycl — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**